src/rest/SearchEndpoint.ts

```diff
@@ -1073,26 +1081,53 @@ export class SearchEndpoint implements ISearchEndpoint {
       }
     });
 
+    const request = () => this.caller.call(params);
+
     try {
-      const response = await this.caller.call(params);
+      const response = await request();
       return response.data as T;
     } catch (error) {
-      if (autoRenewToken && this.accessToken.isExpired(error)) {
-        const renewSuccess = await this.accessToken.doRenew();
-        if (renewSuccess) {
-          return this.performOneCall(params, callOptions, autoRenewToken) as Promise<T>;
-        }
-      } else if (error.statusCode == 0 && this.isRedirecting) {
-        // The page is getting redirected
-        // Set timeout on return with empty string, since it does not really matter
-        _.defer(function() {
-          return '';
-        });
+      const tokenWasRenewed = await this.renewAccessTokenIfExpired(error);
+
+      if (tokenWasRenewed) {
+        return this.performOneCall(params, callOptions) as Promise<T>;
+      }
+
+      if (this.is429Error(error)) {
+        return this.backOff429Request<T>(request);
       }
+
       throw this.handleErrorResponse(error) as any;
     }
   }
 
+  private is429Error(error: IErrorResponse) {
+    return error && error.statusCode === 429;
+  }
+
+  private renewAccessTokenIfExpired(e: IErrorResponse) {
+    return this.accessToken.isExpired(e) && this.accessToken.doRenew();
+  }
+
+  private async backOff429Request<T>(request: () => Promise<ISuccessResponse<{}>>) {
```

Comment:
Same comment about the 429 in the method name, I'd change the naming to "throttled".